yogstation/code/modules/antagonists/darkspawn/darkspawn_abilities/_sacrament.dm

```diff
@@ -1,86 +1,138 @@
 //Turns the darkspawn into a progenitor.
-/datum/action/innate/darkspawn/sacrament
+/datum/action/cooldown/spell/sacrament
 	name = "Sacrament"
-	id = "sacrament"
 	desc = "Ascends into a progenitor. Unless someone else has performed the Sacrament, you must have drained lucidity from 15-30 (check your objective) different people for this to work, and purchased all passive upgrades."
+	panel = "Darkspawn"
+	button_icon = 'yogstation/icons/mob/actions/actions_darkspawn.dmi'
+	background_icon_state = "bg_alien"
+	overlay_icon_state = "bg_alien_border"
+	buttontooltipstyle = "alien"
 	button_icon_state = "sacrament"
 	check_flags =  AB_CHECK_IMMOBILE | AB_CHECK_CONSCIOUS
-	blacklisted = TRUE //baseline
+	spell_requirements = SPELL_REQUIRES_DARKSPAWN
 	var/datum/looping_sound/sacrament/soundloop
+	var/datum/antagonist/darkspawn/darkspawn
+	var/in_use
 
-/datum/action/innate/darkspawn/sacrament/Activate()
-	if(SSticker.mode.sacrament_done)
-		darkspawn.sacrament()
+/datum/action/cooldown/spell/sacrament/Grant(mob/grant_to)
+	. = ..()
+	if(!isdarkspawn(owner))
+		Remove(owner)
+		return
+	darkspawn = isdarkspawn(owner)
+
+/datum/action/cooldown/spell/sacrament/IsAvailable(feedback)
+	if(in_use)
+		if (feedback)
+			owner.balloon_alert(owner, "already in use!")
+		return
+	. = ..()
+
+/datum/action/cooldown/spell/sacrament/cast(atom/cast_on)
+	. = ..()
+	if(!darkspawn)
+		to_chat(owner, span_warning("Error with non darkspawn using sacrament spell"))
 		return
-	if(!darkspawn || darkspawn.lucidity_drained < SSticker.mode.required_succs)
-		to_chat(usr, span_warning("You do not have enough unique lucidity! ([darkspawn.lucidity_drained] / [SSticker.mode.required_succs])"))
+	var/datum/team/darkspawn/team = darkspawn.get_team()
+	if(!team)
 		return
```

Comment:
Maybe return a stack trace here so we can get notified this broke? Or maybe make a new team?